File: jogl/src/examples/java/Shared.java
import com.huskerdev.openglfx.GLCanvasAnimator;
import com.huskerdev.openglfx.OpenGLCanvas;
import com.huskerdev.openglfx.utils.OpenGLFXUtils;
import javafx.application.Application;
import javafx.scene.Scene;
import javafx.scene.control.SplitPane;
import javafx.stage.Stage;
import rendering.ExampleJavaRenderer;

import static com.huskerdev.openglfx.jogl.JOGLFXExecutorKt.JOGL_MODULE;


public class Shared extends Application {

    public static void main(String[] args) {
        System.setProperty("prism.order", "es2");
        System.setProperty("prism.vsync", "false");

        Application.launch(Shared.class);
    }

    public void start(Stage stage) {
        stage.setTitle("Java \"Shared\" example");
        stage.setWidth(400);
        stage.setHeight(400);

        stage.setScene(new Scene(
            new SplitPane(createGL(), createGL())
        ));

        stage.show();
    }

    private OpenGLCanvas createGL() {
        OpenGLCanvas canvas = OpenGLCanvas.create(JOGL_MODULE);
        canvas.setAnimator(new GLCanvasAnimator(60.0));

        canvas.addOnReshapeEvent(ExampleJavaRenderer::reshape);
        canvas.addOnRenderEvent(ExampleJavaRenderer::render);

        return canvas;
    }
}

File: lwjgl/src/examples/kotlin/Shared.kt
import com.huskerdev.openglfx.GLCanvasAnimator
import com.huskerdev.openglfx.GLCanvasAnimator.Companion.UNLIMITED_FPS
import com.huskerdev.openglfx.lwjgl.shared.LWJGLShared
import javafx.application.Application
import javafx.scene.Scene
import javafx.scene.layout.Region
import javafx.stage.Stage
import org.lwjgl.opengl.GL30.*
import rendering.ExampleRenderer
import kotlin.math.sin

fun main(){
    System.setProperty("prism.order", "es2,d3d,sw")
    System.setProperty("prism.vsync", "false")
    Application.launch(SharedExampleApp::class.java)
}

class SharedExampleApp: Application(){

    private lateinit var stage: Stage

    override fun start(stage: Stage?) {
        this.stage = stage!!

        stage.width = 300.0
        stage.height = 300.0
        stage.scene = Scene(createGL())

        stage.show()
    }

    private fun createGL(): Region {
        val canvas = LWJGLShared()
        canvas.animator = GLCanvasAnimator(60.0, started = true)

        canvas.onReshape { ExampleRenderer.reshape(canvas, it) }
        canvas.onRender { ExampleRenderer.render(canvas, it) }

        return canvas
    }
}